src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -19,8 +19,49 @@ public static class ConcurrencyUtilities
         // To maintain SHA-1 backwards compatibility with respect to the length of the hex-encoded hash, the hash will be truncated to a length of 20 bytes.
         private const int HashLength = 20;
         private static readonly TimeSpan SleepDuration = TimeSpan.FromMilliseconds(10);
+        private static Mutex GlobalMutex;
+        private static bool UseDeleteOnClose;
+        private static bool ManualDeleteOnClose;
         private static readonly KeyedLock PerFileLock = new KeyedLock();
 
+        static ConcurrencyUtilities()
+        {
+            // On Non-Windows platforms we use a global mutex for synchronization.
+            //
+            // On Windows opening and locking the file are performed atomically,
+            // and DeleteOnClose deletes the file when there are no more open handles.
+            //
+            // On Unix this locking is not implemented atomically, and DeleteOnClose
+            // deletes ('unlink') the file as soon as the Handle is closed.
+            //
+            // The following example shows how deleting the file in Process 1
+            // allows both Process 2 and Process 3 to obtain a lock because they open
+            // different file entries.
+            //
+            // Process 1    Process 2   Process 3
+            //   open
+            //   lock         open
+            //   unlink
+            //   unlock                   open
+            //   close
+            //                lock        lock
+
+            UseDeleteOnClose = RuntimeEnvironmentHelper.IsWindows;
+            if (!UseDeleteOnClose)
+            {
+                // We need to remain compatible with older SDKS that are still supported.
+                // That means we can really remove lock files when .NET Core 3.1 is EOL on December 3, 2022.
+                // The removing can already be enabled by setting 'NUGET_ConcurrencyUtils_DeleteOnClose' to '1'.
+                // Setting it to 'NoMutex' allows to opt-out of using the GlobalMutex.
+                string NUGET_ConcurrencyUtils_DeleteOnClose = Environment.GetEnvironmentVariable("NUGET_ConcurrencyUtils_DeleteOnClose");
```

Comment:
Please use `EnvironmentVariableWrapper` to enable for better mocking.

Reply:
https://github.com/NuGet/NuGet.Client/blob/680f9bd4e97db7cd7482584276886764de69d3cb/src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs#L55